python/ray/data/dataset.py

```diff
@@ -461,6 +471,15 @@ def map_batches(
                 ``pandas.DataFrame``, "pyarrow" to select ``pyarrow.Table``, or
                 ``"numpy"`` to select ``numpy.ndarray`` for tensor datasets and
                 ``Dict[str, numpy.ndarray]`` for tabular datasets. Default is "default".
+            zero_copy_batch: Whether ``fn`` should be provided zero-copy, read-only
+                batches. If this is ``True`` and no copy is required for the
+                ``batch_format`` conversion, the batch will be a zero-copy, read-only
```

Comment:
Can we expand a bit about this? like 1) what block types support zero-copy (a Core level feature); 2) what block-to-batch conversions are zero-copy. If we have documentations already, we can just link to it.